packages/terra-form/src/Select.jsx

```diff
@@ -62,7 +74,7 @@ const Select = ({
       defaultValue={defaultValue}
       className={classNames('terra-Form-select', additionalSelectFieldProps.className)}
     >
-      {choices.map(choice => <option key={choice.toString()} value={choice}>{choice}</option>)}
```

Comment:
1. To guarantee the unique key for each option, how about using `choice.value + choice.display` instead of just value. Is there probably some special case that same value for different displays? I get an example, as:
```
<option value="----">----
<option value="----">All countries
```
2. Follow the facebook example: https://facebook.github.io/react/docs/lists-and-keys.html#keys, the value of key must be string. If value is number, we'd better to use string interpolation to convert it to string.